include/clang-c/CXCppInterOp.h

```diff
@@ -0,0 +1,643 @@
+
+/**
+ * Declares and executes a code snippet in \c code.
+ *
+ * \param I The interpreter.
+ *
+ * \param code The code snippet to execute.
+ *
+ * \returns a \c CXErrorCode.
+ */
+enum CXErrorCode clang_interpreter_process(CXInterpreter I, const char* code);
+
+/**
+ * An opaque pointer representing a lightweight struct that is used for carrying
+ * execution results.
+ */
+typedef void* CXValue;
+
+/**
+ * Create a CXValue.
+ *
+ * \returns a \c CXValue.
+ */
+CXValue clang_createValue();
+
+/**
+ * Dispose of the given CXValue.
+ *
+ * \param V The CXValue to dispose.
+ */
+void clang_value_dispose(CXValue V);
+
+/**
+ * Declares, executes and stores the execution result to \c V.
+ *
+ * \param[in] I The interpreter.
+ *
+ * \param[in] code The code snippet to evaluate.
+ *
+ * \param[out] V The value to store the execution result.
+ *
+ * \returns a \c CXErrorCode.
+ */
+enum CXErrorCode clang_interpreter_evaluate(CXInterpreter I, const char* code,
+                                            CXValue V);
+
+/**
+ * Looks up the library if access is enabled.
+ *
+ * \param I The interpreter.
+ *
+ * \param lib_name The name of the library to lookup.
+ *
+ * \returns the path to the library.
+ */
+CXString clang_interpreter_lookupLibrary(CXInterpreter I, const char* lib_name);
+
+/**
+ * Finds \c lib_stem considering the list of search paths and loads it by
+ * calling dlopen.
+ *
+ * \param I The interpreter.
+ *
+ * \param lib_stem The stem of the library to load.
+ *
+ * \param lookup Whether to lookup the library or not.
+ *
+ * \returns a \c CXInterpreter_CompilationResult.
+ */
+CXInterpreter_CompilationResult
+clang_interpreter_loadLibrary(CXInterpreter I, const char* lib_stem,
+                              bool lookup);
+
+/**
+ * Finds \c lib_stem considering the list of search paths and unloads it by
+ * calling dlclose.
+ *
+ * \param I The interpreter.
+ *
+ * \param lib_stem The stem of the library to unload.
+ */
+void clang_interpreter_unloadLibrary(CXInterpreter I, const char* lib_stem);
+
+/**
+ * Scans all libraries on the library search path for a given potentially
+ * mangled symbol name.
+ *
+ * \param I The interpreter.
+ *
+ * \param mangled_name The mangled name of the symbol to search for.
+ *
+ * \param search_system Whether to search the system paths or not.
+ *
+ * \returns the path to the first library that contains the symbol definition.
+ */
+CXString clang_interpreter_searchLibrariesForSymbol(CXInterpreter I,
+                                                    const char* mangled_name,
+                                                    bool search_system);
+
+/**
+ * @}
+ */
+
+/**
+ * An opaque pointer representing the object of a given type (\c CXScope).
+ */
+typedef void* CXObject;
+
+/**
+ * \defgroup CPPINTEROP_TYPE_MANIP Type manipulations
+ *
+ * @{
+ */
+
+/**
+ * Describes the kind of entity that a type refers to.
+ */
+enum CXQualTypeKind {
```

Comment:
I am wondering if we should not use `clang/include/clang-c/Index.h` for these?

Reply:
Yes. I think that's possible.

Reply:
ping.